smoketests/__init__.py

```diff
@@ -237,17 +237,16 @@ def stderr_task():
         threading.Thread(target=stderr_task).start()
 
         print("initial update:", proc.stdout.readline())
-        if proc.poll() is not None:
-            if proc.returncode:
-                raise subprocess.CalledProcessError(proc.returncode, fake_args)
-            return lambda timeout=None: []
```

Comment:
This was the source of the race condition - if the process failed quickly enough, we would raise this exception synchronously in the call to `subscribe`. But if the process took a bit longer, then the block below would be the one to raise the exception, which would only be raised to the caller when the caller actually joined the background thread and waited for the subscription to complete.

By removing this block, we now _only_ get exceptions when we wait for the subscription result. We will never get the exception synchronously. This eliminates the race condition - we must always wait for the subscription to complete in order to see the exception.

I have updated the tests accordingly.